[CIS-1049] Add the `Combine` guide for the SDK (#1321)
GetStream/stream-chat-swift: docusaurus/docs/iOS/combine/combine-overview.md

---
title: Overview
---

> The `StreamChat` framework ships with controllers that are fully compatible with `Combine`.

Reviewer: ```suggestion
The `StreamChat` framework ships with `Combine` publishers out of the box. 
```

Author: Done

> You can use these combine-compatible controllers to observe the changes to the controllers, and build your own views to consume the data.

Reviewer: ```suggestion
`Combine` publishers are exposed by `StreamChat` controller classes, these publishers allow you to observe data and to build your own views.
```

Author: Done

Each controller exposes publishers based on the kind of data they control and allow you to observe changes. You can find examples on how to build your own views using these publishers here.

## Using the `Combine` extensions of Controllers

> When using combine extensions of the controllers, you need to call the `synchronize` method on your controller instance before starting to observe the publishers.

Reviewer: Are you sure that this is needed for `Combine` publishers as well?

Author: `Combine` publishers internally use the delegates, and I confirmed that `synchronize` is needed when using them too.
